apps/extension/src/background/approvals/handler.ts

```diff
@@ -96,3 +118,27 @@ const handleRevokeConnectionMsg: (
     return await service.revokeConnection(originToRevoke);
   };
 };
+
+const handleApproveSignArbitraryMsg: (
+  service: ApprovalsService
+) => InternalHandler<ApproveSignArbitraryMsg> = (service) => {
+  return async (_, { signer, data }) => {
+    return await service.approveSignature(signer, data);
+  };
+};
+
+const handleRejectSignatureMsg: (
+  service: ApprovalsService
+) => InternalHandler<RejectSignatureMsg> = (service) => {
+  return async ({ senderTabId: popupTabId }, { msgId }) => {
+    return await service.rejectSignature(popupTabId, msgId);
+  };
+};
+
+const handleSubmitApprovedSignatureMsg: (
+  service: ApprovalsService
+) => InternalHandler<SubmitApprovedSignatureMsg> = (service) => {
+  return async ({ senderTabId: popupTabId }, { msgId, signer }) => {
+    return await service.submitSignature(popupTabId, signer, msgId);
```

Comment:
It looks like the `signer` and `msgId` arguments are the wrong way round.

Reply:
Fixed, thanks!